### ISSUE TITLE
Allow environnement variables to be resolved to lists

### ISSUE THREAD

**Issue by rsemlal** [typesafehub/config#320]
It would be nice to allow environnement variables to be resolved to lists, since in the moment the specification of HOCON explicitly says: "environment variables always become a string value".

A common use case for this is when using Akka Cluster with Docker containers, it would be nice to have the ability to externalize the value of akka.cluster.roles to environnement variables that can be passed when running a docker container, and to do something like that:

```
akka.cluster.roles = [default-role]
akka.cluster.roles = ${?AKKA_ROLE}
```

with AKKA_ROLE an environnement variable that can have the following value: `"role1","role2","role3"` (comma separated list)

At the moment the workaround that is recommended is to add as jvm properties `-Dakka.cluster.roles.0=...`

**Comment by havocp** [comment]
For discussion see #69, #265, #133

The crux of the issue is that a string has to either have syntax or not. If it does, then literal values require quoting/escaping - which for env vars, would often have to be done in shell. So env var values are not treated as having syntax.

**Comment by 2m** [comment]
Fixed in #427
